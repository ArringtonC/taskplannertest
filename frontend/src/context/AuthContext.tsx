import React, {
  createContext,
  useState,
  useContext,
  ReactNode,
  useEffect,
} from 'react';

interface AuthContextType {
  token: string | null;
  login: (token: string) => void;
  logout: () => void;
}

const AuthContext = createContext<AuthContextType>(null!);

interface AuthProviderProps {
  children: ReactNode;
}

export const AuthProvider: React.FC<AuthProviderProps> = ({ children }) => {
  const [token, setToken] = useState<string | null>(() => {
    const savedToken = localStorage.getItem('token');
    console.log(
      'AuthProvider initialized. Token from localStorage:',
      savedToken ? 'present' : 'missing'
    );
    return savedToken;
  });

  // Debug when token changes
  useEffect(() => {
    console.log('Auth token changed:', token ? 'present' : 'missing');
  }, [token]);

  const login = (newToken: string) => {
    console.log('Logging in with new token');
    localStorage.setItem('token', newToken);
    setToken(newToken);
  };

  const logout = () => {
    console.log('Logging out, removing token');
    localStorage.removeItem('token');
    setToken(null);
  };

  return (
    <AuthContext.Provider value={{ token, login, logout }}>
      {children}
    </AuthContext.Provider>
  );
};

export const useAuth = () => useContext(AuthContext);
