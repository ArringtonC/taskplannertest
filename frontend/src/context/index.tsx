import React from 'react';
import { TaskProvider } from './TaskContext';

/**
 * Props for the app provider
 */
interface AppProviderProps {
  children: React.ReactNode;
}

/**
 * Root app provider that combines all context providers
 * @param children React children components
 */
export const AppProvider: React.FC<AppProviderProps> = ({ children }) => {
  return <TaskProvider>{children}</TaskProvider>;
};

// Export all context hooks
export { useTaskContext } from './TaskContext';
