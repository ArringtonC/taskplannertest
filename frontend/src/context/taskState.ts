/**
 * Task state types and initial state
 */

// Task status type
export type TaskStatus =
  | 'pending'
  | 'in-progress'
  | 'completed'
  | 'cancelled'
  | 'deferred';

// Task priority type
export type TaskPriority = 'low' | 'medium' | 'high';

// Task interface
export interface Task {
  _id: string;
  title: string;
  description?: string;
  complexity: number;
  status: TaskStatus;
  priority: TaskPriority;
  dueDate?: string;
  createdBy: string;
  assignedTo?: string;
  project?: string;
  parentTask?: string;
  subtasks?: string[];
  dependencies?: string[];
  tags?: string[];
  createdAt: string;
  updatedAt: string;
}

// Task state interface
export interface TaskState {
  tasks: Task[];
  loading: boolean;
  error: string | null;
  selectedTask: Task | null;
  filteredTasks: Task[];
  filters: {
    status?: TaskStatus;
    priority?: TaskPriority;
    dueDate?: string;
  };
}

// Initial task state
export const initialTaskState: TaskState = {
  tasks: [],
  loading: false,
  error: null,
  selectedTask: null,
  filteredTasks: [],
  filters: {},
};

/**
 * Task action types
 */
export enum TaskActionTypes {
  SET_TASKS = 'SET_TASKS',
  ADD_TASK = 'ADD_TASK',
  UPDATE_TASK = 'UPDATE_TASK',
  DELETE_TASK = 'DELETE_TASK',
  SET_LOADING = 'SET_LOADING',
  SET_ERROR = 'SET_ERROR',
  CLEAR_ERROR = 'CLEAR_ERROR',
  SELECT_TASK = 'SELECT_TASK',
  CLEAR_SELECTED_TASK = 'CLEAR_SELECTED_TASK',
  SET_FILTERS = 'SET_FILTERS',
  CLEAR_FILTERS = 'CLEAR_FILTERS',
  ADD_SUBTASK = 'ADD_SUBTASK',
  ADD_DEPENDENCY = 'ADD_DEPENDENCY',
}

/**
 * Task action interfaces
 */
export interface SetTasksAction {
  type: TaskActionTypes.SET_TASKS;
  payload: Task[];
}

export interface AddTaskAction {
  type: TaskActionTypes.ADD_TASK;
  payload: Task;
}

export interface UpdateTaskAction {
  type: TaskActionTypes.UPDATE_TASK;
  payload: Task;
}

export interface DeleteTaskAction {
  type: TaskActionTypes.DELETE_TASK;
  payload: string; // Task ID
}

export interface SetLoadingAction {
  type: TaskActionTypes.SET_LOADING;
  payload: boolean;
}

export interface SetErrorAction {
  type: TaskActionTypes.SET_ERROR;
  payload: string;
}

export interface ClearErrorAction {
  type: TaskActionTypes.CLEAR_ERROR;
}

export interface SelectTaskAction {
  type: TaskActionTypes.SELECT_TASK;
  payload: Task;
}

export interface ClearSelectedTaskAction {
  type: TaskActionTypes.CLEAR_SELECTED_TASK;
}

export interface SetFiltersAction {
  type: TaskActionTypes.SET_FILTERS;
  payload: {
    status?: TaskStatus;
    priority?: TaskPriority;
    dueDate?: string;
  };
}

export interface ClearFiltersAction {
  type: TaskActionTypes.CLEAR_FILTERS;
}

export interface AddSubtaskAction {
  type: TaskActionTypes.ADD_SUBTASK;
  payload: {
    parentId: string;
    subtask: Task;
  };
}

export interface AddDependencyAction {
  type: TaskActionTypes.ADD_DEPENDENCY;
  payload: {
    taskId: string;
    dependencyId: string;
  };
}

// Union type of all possible task actions
export type TaskAction =
  | SetTasksAction
  | AddTaskAction
  | UpdateTaskAction
  | DeleteTaskAction
  | SetLoadingAction
  | SetErrorAction
  | ClearErrorAction
  | SelectTaskAction
  | ClearSelectedTaskAction
  | SetFiltersAction
  | ClearFiltersAction
  | AddSubtaskAction
  | AddDependencyAction;
