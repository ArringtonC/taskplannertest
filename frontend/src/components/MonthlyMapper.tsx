import React, { useState, useMemo } from 'react';
import { addMonths, subMonths, format, isSameMonth, isBefore, parseISO, getYear, getMonth } from 'date-fns';
import { ArrowLeft, ChevronLeft, ChevronRight } from 'lucide-react';
import { Task } from '../context/taskState';
import { Button } from "@/components/ui/button";

/**
 * Props for MonthlyMapper
 */
export interface MonthlyMapperProps {
  /** List of all tasks to display */
  tasks: Task[];
  /** Maximum number of goals to show (default: 20) */
  maxGoals?: number;
  /** Optional callback for back navigation */
  onBack?: () => void;
  /** Optional initial date for the calendar (for testing or custom start) */
  initialDate?: Date;
}

/**
 * Custom hook to group tasks by goal number for a given month.
 * @param tasks List of tasks
 * @param maxGoals Maximum number of goals
 * @returns Record<goalNumber, Task[]> and "No Rush" bucket
 */
function useTasksByGoal(tasks: Task[], maxGoals: number = 20) {
  // Group tasks by goal number (1-based)
  const grouped: Record<number, Task[]> = {};
  let noRush: Task[] = [];
  for (let i = 1; i <= maxGoals; i++) grouped[i] = [];
  for (const task of tasks) {
    // Assume goal number is encoded in a tag: "goal-<n>"
    const goalTag = task.tags?.find(t => /^goal-\d+$/.test(t));
    if (goalTag) {
      const n = parseInt(goalTag.replace('goal-', ''), 10);
      if (n >= 1 && n <= maxGoals) grouped[n].push(task);
      else noRush.push(task);
    } else {
      noRush.push(task);
    }
  }
  return { grouped, noRush };
}

/**
 * MonthNavigator: Controls for navigating months
 */
interface MonthNavigatorProps {
  currentDate: Date;
  onPrev: () => void;
  onNext: () => void;
}
/**
 * Month navigation controls with accessible buttons
 */
const MonthNavigator: React.FC<MonthNavigatorProps> = ({ currentDate, onPrev, onNext }) => (
  <nav className="flex items-center justify-center mb-4" aria-label="Month navigation">
    <button
      aria-label="Previous month"
      className="p-2 rounded-full hover:bg-gray-700 focus:outline-none focus:ring-2 focus:ring-blue-500"
      onClick={onPrev}
      tabIndex={0}
      type="button"
    >
      <ChevronLeft className="h-5 w-5 text-gray-200" />
    </button>
    <span className="mx-4 text-lg font-semibold text-white" aria-live="polite">
      {format(currentDate, 'MMMM yyyy')}
    </span>
    <button
      aria-label="Next month"
      className="p-2 rounded-full hover:bg-gray-700 focus:outline-none focus:ring-2 focus:ring-blue-500"
      onClick={onNext}
      tabIndex={0}
      type="button"
    >
      <ChevronRight className="h-5 w-5 text-gray-200" />
    </button>
  </nav>
);

/**
 * TaskRow: A single row in the table
 */
interface TaskRowProps {
  goalNumber: number | 'no-rush';
  tasks: Task[];
  month: number;
  year: number;
}
/**
 * Renders a row for a goal (or No Rush) with colored cells for each task
 */
const TaskRow: React.FC<TaskRowProps> = ({ goalNumber, tasks, month, year }) => {
  // Filter tasks for the current month
  const monthTasks = tasks.filter(task => {
    if (!task.dueDate) return false;
    const due = parseISO(task.dueDate);
    return getYear(due) === year && getMonth(due) === month;
  });
  // Overdue: dueDate < now and not completed
  const now = new Date();
  return (
    <tr>
      <th
        scope="row"
        className="px-4 py-2 text-left text-sm font-medium text-gray-200 bg-gray-800"
      >
        {goalNumber === 'no-rush' ? 'No Rush Goals/Personal' : `Goal #${goalNumber}`}
      </th>
      <td className="px-4 py-2">
        <div className="flex flex-wrap gap-2">
          {monthTasks.length === 0 && (
            <span className="text-gray-400 italic">No tasks</span>
          )}
          {monthTasks.map(task => {
            const due = task.dueDate ? parseISO(task.dueDate) : null;
            const overdue = due && isBefore(due, now) && task.status !== 'completed';
            const statusColor = overdue
              ? 'bg-red-600 text-white'
              : task.status === 'completed'
                ? 'bg-green-700 text-white'
                : 'bg-green-900 text-green-200';
            return (
              <span
                key={task._id}
                className={`px-2 py-1 rounded ${statusColor} text-xs font-semibold`}
                title={task.title}
                tabIndex={0}
                aria-label={`${task.title}, status: ${task.status}${overdue ? ', overdue' : ''}`}
              >
                {task.title}
              </span>
            );
          })}
        </div>
      </td>
    </tr>
  );
};

/**
 * TaskTable: Excel-style table for all goals and "No Rush"
 */
interface TaskTableProps {
  grouped: Record<number, Task[]>;
  noRush: Task[];
  month: number;
  year: number;
  maxGoals: number;
}
/**
 * Renders the main table with all goals and "No Rush" row
 */
const TaskTable: React.FC<TaskTableProps> = ({ grouped, noRush, month, year, maxGoals }) => (
  <div className="overflow-x-auto rounded-lg border border-gray-700 bg-gray-900">
    <table className="min-w-full divide-y divide-gray-700" role="table" aria-label="Monthly goals and tasks">
      <thead className="bg-gray-800">
        <tr>
          <th scope="col" className="px-4 py-2 text-left text-xs font-bold text-gray-300 uppercase tracking-wider">
            Goal #
          </th>
          <th scope="col" className="px-4 py-2 text-left text-xs font-bold text-gray-300 uppercase tracking-wider">
            Tasks for Month
          </th>
        </tr>
      </thead>
      <tbody className="divide-y divide-gray-800">
        {Array.from({ length: maxGoals }, (_, i) => (
          <TaskRow
            key={i + 1}
            goalNumber={i + 1}
            tasks={grouped[i + 1]}
            month={month}
            year={year}
          />
        ))}
        <TaskRow
          key="no-rush"
          goalNumber="no-rush"
          tasks={noRush}
          month={month}
          year={year}
        />
      </tbody>
    </table>
  </div>
);

/**
 * MonthlyMapper: Main component
 */
export const MonthlyMapper: React.FC<MonthlyMapperProps> = ({ tasks, maxGoals = 20, onBack, initialDate }) => {
  const [currentDate, setCurrentDate] = useState(() => {
    if (initialDate) return new Date(initialDate.getFullYear(), initialDate.getMonth(), 1);
    const now = new Date();
    return new Date(now.getFullYear(), now.getMonth(), 1);
  });

  // Handlers for month navigation
  const handlePrevMonth = () => setCurrentDate(d => subMonths(d, 1));
  const handleNextMonth = () => setCurrentDate(d => addMonths(d, 1));

  // Derive month/year
  const month = currentDate.getMonth();
  const year = currentDate.getFullYear();

  // Group tasks by goal
  const { grouped, noRush } = useMemo(() => useTasksByGoal(tasks, maxGoals), [tasks, maxGoals]);

  return (
    <section className="max-w-5xl mx-auto p-4">
      <div className="flex items-center mb-6">
        <button
          onClick={onBack}
          className="flex items-center text-gray-300 hover:text-white focus:outline-none focus:ring-2 focus:ring-blue-500 mr-4"
          aria-label="Back to Dashboard"
          type="button"
        >
          <ArrowLeft className="h-5 w-5 mr-1" />
          Back to Dashboard
        </button>
        <h1 className="text-2xl font-bold text-white flex-1 text-center" tabIndex={0} aria-label="Monthly Mapper Heading">
          Monthly Mapper
        </h1>
      </div>
      <MonthNavigator currentDate={currentDate} onPrev={handlePrevMonth} onNext={handleNextMonth} />
      <TaskTable grouped={grouped} noRush={noRush} month={month} year={year} maxGoals={maxGoals} />
    </section>
  );
};

/**
 * README: Usage Example
 *
 * ```tsx
 * import { MonthlyMapper } from './components/MonthlyMapper';
 * import { Task } from './context/taskState';
 *
 * // Example usage in a page/component:
 * const tasks: Task[] = [...];
 *
 * <MonthlyMapper tasks={tasks} maxGoals={20} onBack={() => navigate('/dashboard')} initialDate={new Date(2024, 4, 1)} />
 * ```
 *
 * - `tasks`: Array of Task objects (see Task interface in context/taskState.ts)
 * - `maxGoals`: (optional) Maximum number of goals to display (default 20)
 * - `onBack`: (optional) Callback for the back button
 * - `initialDate`: (optional) Initial date for the calendar (useful for tests)
 *
 * The component is fully accessible, responsive, and styled for a dark theme with Tailwind CSS.
 *
 * ## Task Grouping
 * Tasks are grouped by goal number using a `goal-<n>` tag (e.g., `goal-1`, `goal-2`). Tasks without a goal tag are placed in the "No Rush Goals/Personal" row.
 *
 * ## Testing
 * - Unit test date navigation and grouping logic
 * - Snapshot test the rendered table
 */ 