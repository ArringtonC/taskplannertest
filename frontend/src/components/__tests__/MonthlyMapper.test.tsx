import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { MonthlyMapper } from '../MonthlyMapper';
import { Task } from '../../context/taskState';
import { addMonths, subMonths } from 'date-fns';

describe('MonthlyMapper', () => {
  const baseDate = new Date(2024, 4, 1); // May 2024
  const tasks: Task[] = [
    {
      _id: '1',
      title: 'Goal 1 Task',
      complexity: 3,
      status: 'pending',
      priority: 'high',
      dueDate: '2024-05-15T00:00:00.000Z',
      createdBy: 'user',
      createdAt: '2024-04-01T00:00:00.000Z',
      updatedAt: '2024-04-01T00:00:00.000Z',
      tags: ['goal-1'],
    },
    {
      _id: '2',
      title: 'Goal 2 Completed',
      complexity: 2,
      status: 'completed',
      priority: 'medium',
      dueDate: '2024-05-10T00:00:00.000Z',
      createdBy: 'user',
      createdAt: '2024-04-01T00:00:00.000Z',
      updatedAt: '2024-04-01T00:00:00.000Z',
      tags: ['goal-2'],
    },
    {
      _id: '3',
      title: 'No Rush Task',
      complexity: 1,
      status: 'pending',
      priority: 'low',
      dueDate: '2024-05-20T00:00:00.000Z',
      createdBy: 'user',
      createdAt: '2024-04-01T00:00:00.000Z',
      updatedAt: '2024-04-01T00:00:00.000Z',
      tags: [],
    },
    {
      _id: '4',
      title: 'Overdue Task',
      complexity: 2,
      status: 'pending',
      priority: 'medium',
      dueDate: '2024-05-01T00:00:00.000Z',
      createdBy: 'user',
      createdAt: '2023-04-01T00:00:00.000Z',
      updatedAt: '2023-04-01T00:00:00.000Z',
      tags: ['goal-1'],
    },
  ];

  it('renders the heading and navigation', () => {
    render(<MonthlyMapper tasks={tasks} initialDate={new Date(2024, 4, 1)} />);
    expect(screen.getByRole('heading', { name: /monthly mapper/i })).toBeInTheDocument();
    expect(screen.getByLabelText(/back to dashboard/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/month navigation/i)).toBeInTheDocument();
  });

  it('shows tasks grouped by goal and "No Rush"', () => {
    render(<MonthlyMapper tasks={tasks} initialDate={new Date(2024, 4, 1)} />);
    expect(screen.getByText('Goal #1')).toBeInTheDocument();
    expect(screen.getByText('Goal #2')).toBeInTheDocument();
    expect(screen.getByText('No Rush Goals/Personal')).toBeInTheDocument();
    expect(screen.getByText('Goal 1 Task')).toBeInTheDocument();
    expect(screen.getByText('Goal 2 Completed')).toBeInTheDocument();
    expect(screen.getByText('No Rush Task')).toBeInTheDocument();
  });

  it('applies correct styling for overdue and completed tasks', () => {
    render(<MonthlyMapper tasks={tasks} initialDate={new Date(2024, 4, 1)} />);
    const overdue = screen.getByText('Goal 1 Task');
    expect(overdue).toHaveClass('bg-red-600');
    const completed = screen.getByText('Goal 2 Completed');
    expect(completed).toHaveClass('bg-green-700');
  });

  it('navigates months with prev/next buttons', () => {
    render(<MonthlyMapper tasks={tasks} initialDate={new Date(2024, 4, 1)} />);
    const monthLabel = screen.getByText(/may 2024/i);
    expect(monthLabel).toBeInTheDocument();
    const nextBtn = screen.getByLabelText(/next month/i);
    fireEvent.click(nextBtn);
    expect(screen.getByText(/june 2024/i)).toBeInTheDocument();
    const prevBtn = screen.getByLabelText(/previous month/i);
    fireEvent.click(prevBtn);
    fireEvent.click(prevBtn);
    expect(screen.getByText(/april 2024/i)).toBeInTheDocument();
  });

  it('calls onBack when back button is clicked', () => {
    const onBack = jest.fn();
    render(<MonthlyMapper tasks={tasks} onBack={onBack} initialDate={new Date(2024, 4, 1)} />);
    fireEvent.click(screen.getByLabelText(/back to dashboard/i));
    expect(onBack).toHaveBeenCalled();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<MonthlyMapper tasks={tasks} initialDate={new Date(2024, 4, 1)} />);
    expect(asFragment()).toMatchSnapshot();
  });
}); 