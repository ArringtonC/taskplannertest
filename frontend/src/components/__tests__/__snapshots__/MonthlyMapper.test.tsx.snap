// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonthlyMapper matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="max-w-5xl mx-auto p-4"
  >
    <div
      class="flex items-center mb-6"
    >
      <button
        aria-label="Back to Dashboard"
        class="flex items-center text-gray-300 hover:text-white focus:outline-none focus:ring-2 focus:ring-blue-500 mr-4"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-arrow-left h-5 w-5 mr-1"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m12 19-7-7 7-7"
          />
          <path
            d="M19 12H5"
          />
        </svg>
        Back to Dashboard
      </button>
      <h1
        aria-label="Monthly Mapper Heading"
        class="text-2xl font-bold text-white flex-1 text-center"
        tabindex="0"
      >
        Monthly Mapper
      </h1>
    </div>
    <nav
      aria-label="Month navigation"
      class="flex items-center justify-center mb-4"
    >
      <button
        aria-label="Previous month"
        class="p-2 rounded-full hover:bg-gray-700 focus:outline-none focus:ring-2 focus:ring-blue-500"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-chevron-left h-5 w-5 text-gray-200"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m15 18-6-6 6-6"
          />
        </svg>
      </button>
      <span
        aria-live="polite"
        class="mx-4 text-lg font-semibold text-white"
      >
        May 2024
      </span>
      <button
        aria-label="Next month"
        class="p-2 rounded-full hover:bg-gray-700 focus:outline-none focus:ring-2 focus:ring-blue-500"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-chevron-right h-5 w-5 text-gray-200"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m9 18 6-6-6-6"
          />
        </svg>
      </button>
    </nav>
    <div
      class="overflow-x-auto rounded-lg border border-gray-700 bg-gray-900"
    >
      <table
        aria-label="Monthly goals and tasks"
        class="min-w-full divide-y divide-gray-700"
        role="table"
      >
        <thead
          class="bg-gray-800"
        >
          <tr>
            <th
              class="px-4 py-2 text-left text-xs font-bold text-gray-300 uppercase tracking-wider"
              scope="col"
            >
              Goal #
            </th>
            <th
              class="px-4 py-2 text-left text-xs font-bold text-gray-300 uppercase tracking-wider"
              scope="col"
            >
              Tasks for Month
            </th>
          </tr>
        </thead>
        <tbody
          class="divide-y divide-gray-800"
        >
          <tr>
            <th
              class="px-4 py-2 text-left text-sm font-medium text-gray-200 bg-gray-800"
              scope="row"
            >
              Goal #1
            </th>
            <td
              class="px-4 py-2"
            >
              <div
                class="flex flex-wrap gap-2"
              >
                <span
                  aria-label="Goal 1 Task, status: pending, overdue"
                  class="px-2 py-1 rounded bg-red-600 text-white text-xs font-semibold"
                  tabindex="0"
                  title="Goal 1 Task"
                >
                  Goal 1 Task
                </span>
              </div>
            </td>
          </tr>
          <tr>
            <th
              class="px-4 py-2 text-left text-sm font-medium text-gray-200 bg-gray-800"
              scope="row"
            >
              Goal #2
            </th>
            <td
              class="px-4 py-2"
            >
              <div
                class="flex flex-wrap gap-2"
              >
                <span
                  aria-label="Goal 2 Completed, status: completed"
                  class="px-2 py-1 rounded bg-green-700 text-white text-xs font-semibold"
                  tabindex="0"
                  title="Goal 2 Completed"
                >
                  Goal 2 Completed
                </span>
              </div>
            </td>
          </tr>
          <tr>
            <th
              class="px-4 py-2 text-left text-sm font-medium text-gray-200 bg-gray-800"
              scope="row"
            >
              Goal #3
            </th>
            <td
              class="px-4 py-2"
            >
              <div
                class="flex flex-wrap gap-2"
              >
                <span
                  class="text-gray-400 italic"
                >
                  No tasks
                </span>
              </div>
            </td>
          </tr>
          <tr>
            <th
              class="px-4 py-2 text-left text-sm font-medium text-gray-200 bg-gray-800"
              scope="row"
            >
              Goal #4
            </th>
            <td
              class="px-4 py-2"
            >
              <div
                class="flex flex-wrap gap-2"
              >
                <span
                  class="text-gray-400 italic"
                >
                  No tasks
                </span>
              </div>
            </td>
          </tr>
          <tr>
            <th
              class="px-4 py-2 text-left text-sm font-medium text-gray-200 bg-gray-800"
              scope="row"
            >
              Goal #5
            </th>
            <td
              class="px-4 py-2"
            >
              <div
                class="flex flex-wrap gap-2"
              >
                <span
                  class="text-gray-400 italic"
                >
                  No tasks
                </span>
              </div>
            </td>
          </tr>
          <tr>
            <th
              class="px-4 py-2 text-left text-sm font-medium text-gray-200 bg-gray-800"
              scope="row"
            >
              Goal #6
            </th>
            <td
              class="px-4 py-2"
            >
              <div
                class="flex flex-wrap gap-2"
              >
                <span
                  class="text-gray-400 italic"
                >
                  No tasks
                </span>
              </div>
            </td>
          </tr>
          <tr>
            <th
              class="px-4 py-2 text-left text-sm font-medium text-gray-200 bg-gray-800"
              scope="row"
            >
              Goal #7
            </th>
            <td
              class="px-4 py-2"
            >
              <div
                class="flex flex-wrap gap-2"
              >
                <span
                  class="text-gray-400 italic"
                >
                  No tasks
                </span>
              </div>
            </td>
          </tr>
          <tr>
            <th
              class="px-4 py-2 text-left text-sm font-medium text-gray-200 bg-gray-800"
              scope="row"
            >
              Goal #8
            </th>
            <td
              class="px-4 py-2"
            >
              <div
                class="flex flex-wrap gap-2"
              >
                <span
                  class="text-gray-400 italic"
                >
                  No tasks
                </span>
              </div>
            </td>
          </tr>
          <tr>
            <th
              class="px-4 py-2 text-left text-sm font-medium text-gray-200 bg-gray-800"
              scope="row"
            >
              Goal #9
            </th>
            <td
              class="px-4 py-2"
            >
              <div
                class="flex flex-wrap gap-2"
              >
                <span
                  class="text-gray-400 italic"
                >
                  No tasks
                </span>
              </div>
            </td>
          </tr>
          <tr>
            <th
              class="px-4 py-2 text-left text-sm font-medium text-gray-200 bg-gray-800"
              scope="row"
            >
              Goal #10
            </th>
            <td
              class="px-4 py-2"
            >
              <div
                class="flex flex-wrap gap-2"
              >
                <span
                  class="text-gray-400 italic"
                >
                  No tasks
                </span>
              </div>
            </td>
          </tr>
          <tr>
            <th
              class="px-4 py-2 text-left text-sm font-medium text-gray-200 bg-gray-800"
              scope="row"
            >
              Goal #11
            </th>
            <td
              class="px-4 py-2"
            >
              <div
                class="flex flex-wrap gap-2"
              >
                <span
                  class="text-gray-400 italic"
                >
                  No tasks
                </span>
              </div>
            </td>
          </tr>
          <tr>
            <th
              class="px-4 py-2 text-left text-sm font-medium text-gray-200 bg-gray-800"
              scope="row"
            >
              Goal #12
            </th>
            <td
              class="px-4 py-2"
            >
              <div
                class="flex flex-wrap gap-2"
              >
                <span
                  class="text-gray-400 italic"
                >
                  No tasks
                </span>
              </div>
            </td>
          </tr>
          <tr>
            <th
              class="px-4 py-2 text-left text-sm font-medium text-gray-200 bg-gray-800"
              scope="row"
            >
              Goal #13
            </th>
            <td
              class="px-4 py-2"
            >
              <div
                class="flex flex-wrap gap-2"
              >
                <span
                  class="text-gray-400 italic"
                >
                  No tasks
                </span>
              </div>
            </td>
          </tr>
          <tr>
            <th
              class="px-4 py-2 text-left text-sm font-medium text-gray-200 bg-gray-800"
              scope="row"
            >
              Goal #14
            </th>
            <td
              class="px-4 py-2"
            >
              <div
                class="flex flex-wrap gap-2"
              >
                <span
                  class="text-gray-400 italic"
                >
                  No tasks
                </span>
              </div>
            </td>
          </tr>
          <tr>
            <th
              class="px-4 py-2 text-left text-sm font-medium text-gray-200 bg-gray-800"
              scope="row"
            >
              Goal #15
            </th>
            <td
              class="px-4 py-2"
            >
              <div
                class="flex flex-wrap gap-2"
              >
                <span
                  class="text-gray-400 italic"
                >
                  No tasks
                </span>
              </div>
            </td>
          </tr>
          <tr>
            <th
              class="px-4 py-2 text-left text-sm font-medium text-gray-200 bg-gray-800"
              scope="row"
            >
              Goal #16
            </th>
            <td
              class="px-4 py-2"
            >
              <div
                class="flex flex-wrap gap-2"
              >
                <span
                  class="text-gray-400 italic"
                >
                  No tasks
                </span>
              </div>
            </td>
          </tr>
          <tr>
            <th
              class="px-4 py-2 text-left text-sm font-medium text-gray-200 bg-gray-800"
              scope="row"
            >
              Goal #17
            </th>
            <td
              class="px-4 py-2"
            >
              <div
                class="flex flex-wrap gap-2"
              >
                <span
                  class="text-gray-400 italic"
                >
                  No tasks
                </span>
              </div>
            </td>
          </tr>
          <tr>
            <th
              class="px-4 py-2 text-left text-sm font-medium text-gray-200 bg-gray-800"
              scope="row"
            >
              Goal #18
            </th>
            <td
              class="px-4 py-2"
            >
              <div
                class="flex flex-wrap gap-2"
              >
                <span
                  class="text-gray-400 italic"
                >
                  No tasks
                </span>
              </div>
            </td>
          </tr>
          <tr>
            <th
              class="px-4 py-2 text-left text-sm font-medium text-gray-200 bg-gray-800"
              scope="row"
            >
              Goal #19
            </th>
            <td
              class="px-4 py-2"
            >
              <div
                class="flex flex-wrap gap-2"
              >
                <span
                  class="text-gray-400 italic"
                >
                  No tasks
                </span>
              </div>
            </td>
          </tr>
          <tr>
            <th
              class="px-4 py-2 text-left text-sm font-medium text-gray-200 bg-gray-800"
              scope="row"
            >
              Goal #20
            </th>
            <td
              class="px-4 py-2"
            >
              <div
                class="flex flex-wrap gap-2"
              >
                <span
                  class="text-gray-400 italic"
                >
                  No tasks
                </span>
              </div>
            </td>
          </tr>
          <tr>
            <th
              class="px-4 py-2 text-left text-sm font-medium text-gray-200 bg-gray-800"
              scope="row"
            >
              No Rush Goals/Personal
            </th>
            <td
              class="px-4 py-2"
            >
              <div
                class="flex flex-wrap gap-2"
              >
                <span
                  aria-label="No Rush Task, status: pending, overdue"
                  class="px-2 py-1 rounded bg-red-600 text-white text-xs font-semibold"
                  tabindex="0"
                  title="No Rush Task"
                >
                  No Rush Task
                </span>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </section>
</DocumentFragment>
`;
